### PR TITLE
feat: remove all TypeScript references when `isTypeScript` is passed as `false` to `remix.init`

### LINKED ISSUE #61: Oops, remix.init failed when choosing Javascript instead of Typescript
### Have you experienced this bug with the latest version of the template?

yes

### Steps to Reproduce

Running: `npx create-remix --template remix-run/indie-stack` gives an error when selecting the Javascript option instead of the Typescript option:

```
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": SQLite database "data.db" at "file:./data.db?connection_limit=1"

SQLite database data.db created at file:./data.db?connection_limit=1

Applying migration `20220307190657_init`

The following migration(s) have been applied:

migrations/
  └─ 20220307190657_init/
    └─ migration.sql

Your database is now in sync with your schema.

✔ Generated Prisma Client (3.11.1 | library) to ./node_modules/@prisma/client in
 244ms


Running seed command `ts-node --require tsconfig-paths/register prisma/seed.ts` ...
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module './seed.ts'
Require stack:
- /Users/jan/Sites/remix-run/prisma/imaginaryUncacheableRequireRes
```

### Expected Behavior

I'd expect the `seed.ts` being a  `.jsx` file when using the Javascript option. Think this is coming from the `package.json` script:

```
"seed": "ts-node --require tsconfig-paths/register prisma/seed.ts"
```
Think basically the script should be changed to use the `prisma/seed.js` file when using the Javascript option. That file is present.

### Actual Behavior

The `seed.ts` file is not present so the `seed`script fails. There is however a `prisma/seed.js` file available.

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
- [x] Update `prisma seed` script to use JS
- [x] Cleanup `include` from `jsconfig`
- [x] Replace `vitest` config with JS version
- [x] Replace `vitest-tsconfig-paths` with js equivalent
- [x] Updates `cypress`' `cy.exec` calls to use JS
- [x] Replace mocks `index` with JS version

---

https://github.com/remix-run/remix/pull/3150 needs to be merged first

Closes #61

### CODE REVIEW

**Comment by jan-dh** [comment]
Not sure but for me to get this to work I have to:
1. Update the package.json to this (using .mjs format):
`"seed": "node prisma/seed.mjs"`

2. In the prisma/seed.mjs  file:
change this:
`import { PrismaClient } from "@prisma/client";`
to 
```
import pkg from "@prisma/client";
const prisma = new pkg.PrismaClient();
```

File: package.json [hunk 1]
```diff
@@ -76,6 +76,7 @@
     "tsconfig-paths": "^3.14.1",
     "typescript": "^4.6.3",
     "vite": "^2.9.6",
+    "vite-jsconfig-paths": "^2.0.1",
```
Comment (MichaelDeBoey) [inline package.json:79]: We'll keep using this until `vite-tsconfig-paths` is updated to use latest version of `tsconfig-paths` (which will support `jsconfig.json` files (see https://github.com/dividab/tsconfig-paths/pull/199)) once it's released

File: remix.init/index.js [hunk 2]
```diff
@@ -2,26 +2,87 @@ const { execSync } = require("child_process");
...
-const sort = require("sort-package-json");
+const PackageJson = require("@npmcli/package-json");
+const inquirer = require("inquirer");
+
+const cleanupCypressFiles = (filesEntries) =>
+  filesEntries.flatMap(([filePath, content]) => {
+    const newContent = content
+      .replace(
+        "npx ts-node --require tsconfig-paths/register ./cypress/support/create-user.ts",
+        "node --require tsconfig-paths/register ./cypress/support/create-user.js"
```
Comment (MichaelDeBoey) [inline remix.init/index.js:15]: `tsconfig-paths` with `jsconfig.json` support isn't supported yet, so we need to figure out a way to register `~` paths in the meantime

File: package.json (hunk 1 above)
Comment (MichaelDeBoey) [reply package.json:79]: Created https://github.com/aleclarson/vite-tsconfig-paths/issues/51 to track this in `vite-tsconfig-paths`

File: remix.init/index.js (hunk 2 above)
Comment (MichaelDeBoey) [reply remix.init/index.js:15]: `jsconfig.json` support in `tsconfig-paths` is now released

https://github.com/dividab/tsconfig-paths/blob/master/CHANGELOG.md#400---2022-05-02

File: package.json (hunk 1 above)
Comment (MichaelDeBoey) [reply package.json:79]: https://github.com/aleclarson/vite-tsconfig-paths/pull/57 is now merged and released in [v3.5.0](https://github.com/aleclarson/vite-tsconfig-paths/releases/tag/v3.5.0), so we can remove `vite-jsconfig-paths` 🎉

**Comment by kentcdodds** [comment]
I handled this. We can probably make improvements when that PR is merged though. Let's close this one for now though

**Comment by MichaelDeBoey** [comment]
@kentcdodds I rebased my changes onto latest `main`.

Re-opening this as this can now be reviewed again.

**Comment by kentcdodds** [comment]
I think it's possible that Cypress is being a bit flakey 😩

**Comment by kentcdodds** [comment]
Yep. That's what I call flake. We're probably doing something wrong/suboptimal, but I don't have time right now to dig into what it is...

**Comment by MichaelDeBoey** [comment]
Do we first want to fix the flackyness or do we merge this already?

**Comment by kentcdodds** [comment]
Oh dang! I didn't realize that the other PR had been merged. Thank you!

**Review by kentcdodds — APPROVED** [review]
Nice! Thanks for working so hard on this!

**Comment by meandillar** [comment]
> Not sure but for me to get this to work I have to:
> 
> 1. Update the package.json to this (using .mjs format):
>    `"seed": "node prisma/seed.mjs"`
> 2. In the prisma/seed.mjs  file:
>    change this:
>    `import { PrismaClient } from "@prisma/client";`
>    to
> 
> ```
> import pkg from "@prisma/client";
> const prisma = new pkg.PrismaClient();
> ```

My installation failed for the Indie stack with the latest remix version and the above (changing seed.js to seed.mjs) fixed it.

Here was the error:

```
Running seed command `node --require tsconfig-paths/register prisma/seed.js` ...
TypeError: Cannot read property 'hash' of undefined
    at seed
```

**Pull request merged**